inspirehep/modules/search/api.py

```diff
@@ -77,12 +89,32 @@ def mget(self, uuids, **kwargs):
         return [document['_source'] for document in documents['docs']]
 
 
+def inspire_filter():
+    """Filter applied to all queries."""
+    if request:
+        collection = request.values.get('cc', 'Literature')
+
+        user_roles = [r.name for r in current_user.roles]
+        if 'superuser' in user_roles:
+            user_coll = all_restricted_collections
+        else:
+            user_coll = user_collections
+
+        query = Q('match', _collections=collection)
+
+        for collection in list(all_restricted_collections - user_coll):
+            query = query & ~Q('match', _collections=collection)
```

Comment:
The query returned here is added to a ES filter query. You can see the output in this test https://github.com/inspirehep/inspire-next/pull/1589/files#diff-8eedb14f28e016fe0e021789e25878abR336